blog/2025/2025-05-09-blender-baking/index.md

```diff
@@ -0,0 +1,222 @@
+
+![Select Normal Node](select-normal-node.webp)
+
+Now select the normal map texture node in the shader nodes for the current material, this tells blender to use it as the destination for baking.
+
+![Hit Bake](hit-bake.webp)
+
+We can finally hit bake!
+
+![Messed Up Bake](messed-up-bake.webp)
+
+After some amount of processing time, you should see a preview of the normal map. There is also a 99% chance it will be messed up in some way.
+
+![Messed Up Bake Normals](messed-up-bake-normals.webp)
+
+As you can see looking at our "Low Poly" object something is very off.
+
+![Extrusion And Max Ray Distance](extrusion-and-max-ray-distance.webp)
+
+The solution to this problem is adjusting two very important parameters for baking. They are "Extrusion" and "Max Ray Distance". 
+
+In Blender baking works by shooting out rays from the Bake Target. Since our Low poly mesh doesn't lie completely outside the surface of the the high poly object we blender needs to effectively extrude the surfaces of the target outward so that the high poly object is completely contained within the low poly one. The amount that it does this is the "Extrusion" and the length of the rays are "Max Ray Distance".
+
+Now of course you are probably wondering at this point, how do I know what to set these numbers to? My rule of thumb is to set extrusion to the smallest value you can that makes the green pixels in the normal map go away. Then set the Max Ray Distance to ~1.5-2 times the Extrusion. 
+
+![Good Values](good-values.webp)
+
+In this case 0.1 and 0.2 are good values.
+
+![Not Enough Distance](not-enough-distance.webp)
+
+If the Max Ray Distance was too low, eg. 0.1, we would get holes in our normal map as shown above.
+
+![Perfect Normals](perfect-normals.webp)
+
+If our values are set properly we get a nice normal map without any artifacts.
+
+![Bumpy Bowl](bumpy-bowl.webp)
+
+We can also now look at our "Low Poly" object and see that it looks nice and bumpy. But there is one tiny problem, It's far too shiny! This is because its roughness map is entirely black or 0.0, this corresponds to a mirror like shine. So of course our next step should be to bake a roughness map
+
+# Baking a Roughness Map
+
+![Select Roughness](select-roughness.webp)
+
+With your selection still on the high poly and your active still on the bake target select the roughness map texture node in the shader nodes editor.
+
+![Bake Type Rough](bake-type-rough.webp)
+
+Select "Roughness" for Bake Type and hit Bake again.
+
+![Roughness Result](roughness-result.webp)
+
+After waiting for the bake to complete we now have a roughness map and the shininess of our bowl now looks correct. Last but certainly not least we need to bake albedo, this is the actual surface colour of our object.
```

Comment:
...shininess of our bowl looks correct

now twice in that sentence sounds weird